<launch>
	<!-- Load dummy URDF -->
	<param name="robot_description" command="$(find xacro)/xacro.py '$(find cwru_davinci_traj_streamer)/urdf/dummy_psm.xacro'" />

	<node name="davinci_traj_controller_1" pkg="cwru_davinci_traj_streamer" type="davinci_traj_controller" respawn="false" output="screen" ns="/dv_control/PSM1" args="PSM1"/>
	
	<!-- Load joint controller configurations from YAML file to parameter server -->
	<rosparam file="$(find cwru_davinci_traj_streamer)/config/control_1.yaml" command="load"/>
	<!-- load the controllers -->
	<node
		name="controller_spawner" 
		pkg="controller_manager"
		type="spawner"
		respawn="false"
		output="screen"
		ns="/dv_control/PSM1"
		args="joint_state_controller joint_traj_controller"
	/>
	
	<!-- Load joint controller configurations from YAML file to parameter server -->
	<rosparam file="$(find cwru_davinci_traj_streamer)/config/control_2.yaml" command="load"/>
	
	<node name="davinci_traj_controller_2" pkg="cwru_davinci_traj_streamer" type="davinci_traj_controller" respawn="false" output="screen" ns="/dv_control/PSM2" args="PSM2"/>
	<!-- load the controllers -->
	<node
		name="controller_spawner" 
		pkg="controller_manager"
		type="spawner"
		respawn="false"
		output="screen"
		ns="/dv_control/PSM2"
		args="joint_state_controller joint_traj_controller"
	/>	  
</launch>
